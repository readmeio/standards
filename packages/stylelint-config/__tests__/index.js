const config = require('..');
const fs = require('fs');
const stylelint = require('stylelint');

const invalidScss = fs.readFileSync('./__tests__/invalid.scss', 'utf-8');
const validScss = fs.readFileSync('./__tests__/valid.scss', 'utf-8');

describe('stylelint-config', () => {
  let data;

  describe('with a valid file', () => {
    beforeEach(async () => {
      data = await stylelint.lint({
        code: validScss,
        config,
      });
    });

    it('has no errors', () => {
      expect(data.errored).toBeFalsy();
    });

    it('flags no warnings', () => {
      expect(data.results[0].warnings).toHaveLength(0);
    });
  });

  describe('with an invalid file and auto-fix enabled', () => {
    beforeEach(async () => {
      data = await stylelint.lint({
        code: invalidScss,
        config,
        fix: true,
      });
    });

    it('matches the auto-fixed snapshot', () => {
      expect(data.output).toMatchSnapshot();
    });

    it('has errors', () => {
      expect(data.errored).toBeTruthy();
    });

    it('flags warnings', () => {
      expect(data.results[0].warnings).toHaveLength(4);
    });
  });
});
