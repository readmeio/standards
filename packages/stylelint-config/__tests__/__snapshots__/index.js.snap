// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelint-config with an invalid file and auto-fix enabled matches the auto-fixed snapshot 1`] = `
"@import url(\\"x.css\\");
@import url(\\"y.css\\");

/**
 * Multi-line comment
 */

:root {
  --brand-red: hsl(5deg 10% 40%);
}

.selector-1,
.selector-2,
.selector-3[type=\\"text\\"] {
  background: linear-gradient(#fff, rgb(0 0 0 / 80%));
  box-sizing: border-box;
  display: block;
  color: var(--brand-red);
}

.selector-a,
.selector-b:not(:first-child) {
  padding: 10px !important;
  top: calc(100% - 2rem);
}

.selector-x {
  width: 10%;
}

.selector-y {
  width: 20%;
}

.selector-z {
  width: 30%;
}

/* Single-line comment */

@media (width >= 60em) {
  .selector {
    /* Flush to parent comment */
    transform: translate(1, 1) scale(3);
  }
}

@media (orientation: portrait), projection and (color) {
  .selector-i + .selector-ii {
    background: hsl(20deg 25% 33%);
    font-family: Helvetica, \\"Arial Black\\", sans-serif;
  }
}

/* Flush single line comment */
@media screen and (min-resolution: 192dpi), screen and (min-resolution: 2dppx) {
  .selector {
    animation: 3s none fade-in;
    background-image: repeating-linear-gradient(
      -45deg,
      transparent,
      #fff 25px,
      rgb(255 255 255 / 100%) 50px
    );
    margin: 10px;
    margin-bottom: 5px;
    box-shadow: 0 1px 1px #000, 0 1px 0 #fff, 2px 2px 1px 1px #ccc inset;
    height: 10rem;
  }

  /* Flush nested single line comment */
  .selector::after {
    content: \\"â†’\\";
    background-image: url(\\"x.svg\\");
  }
}

@keyframes fade-in {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}
"
`;
